### PR TITLE
iotlab: node selection via Makefile

### PR DESCRIPTION
**Pull request by OlegHahm**
Flashing and serial access of a subset of reserved nodes is not possible via make. This PR adds this functionality.

### CODE REVIEW

**Comment by cgundogan** [comment]
`iotlab-exp` fails:
```
usage: experiment-cli get [-h] [-i EXPERIMENT_ID]
                          (-r | -ri | -s | -st | -p | -a | -l)
                          [--offset OFFSET] [--limit LIMIT] [--state STATE]
experiment-cli get: error: argument -i/--id: expected one argument
usage: experiment-cli submit [-h] -l NODES_LIST [-n NAME] -d DURATION
                             [-r RESERVATION] [-p]
experiment-cli submit: error: argument -l/--list: Unknown site name ''
usage: experiment-cli wait [-h] [-i EXPERIMENT_ID] [--state STATE]
                           [--step STEP] [--timeout TIMEOUT]
experiment-cli wait: error: argument -i/--id: expected one argument
```
The problem is that `experiment-cli get -i $(IOTLAB_EXP_ID) --resources ...` is invoked before submitting the experiment => there is no `$(IOTLAB_EXP_ID)` yet.

File: dist/testbed-support/Makefile.iotlab [hunk 1]
```diff
@@ -62,4 +63,8 @@ iotlab-debug-server: $(IOTLAB_AUTH)
 
 iotlab-term:
 	$(AD)ssh -t $(IOTLAB_AUTHORITY) "test -f ~/.iotlabrc || auth-cli -u $(IOTLAB_USER)"
-	$(AD)ssh -t $(IOTLAB_AUTHORITY) "serial_aggregator -i $(IOTLAB_EXP_ID)"
+    ifndef NODES_PARAM
```
Comment (cgundogan) [inline dist/testbed-support/Makefile.iotlab]: I would prefer a blank line above this line

File: dist/testbed-support/Makefile.iotlab [hunk 2]
```diff
@@ -1,4 +1,4 @@
-.PHONY: iotlab-auth iotlab-exp iotlab-flash iotlab-reset iotlab-term
+.PHONY: iotlab-auth iotlab-exp iotlab-flash iotlab-reset iotlab-term check-exp
```
Comment (cgundogan) [inline dist/testbed-support/Makefile.iotlab]: please prefix `check-exp` with `iotlab-` or something else that is not so obvious

**Comment by cgundogan** [comment]
ACK after addressing my PR

**Comment by cgundogan** [comment]
and .. dear @OlegHahm, please squash!

**Comment by OlegHahm** [comment]
squashed

**Comment by authmillenon** [comment]
Did you rebase?

**Pull request merged**